src/circuit/gadget/sinsemilla/chip.rs

```diff
@@ -182,35 +192,30 @@ impl SinsemillaChip {
             // Y_A = (lambda_1 + lambda_2) * (x_a - x_r)
             let Y_A_next = Y_A(meta, Rotation::next());
 
-            // Check that the initial x_A, x_P, lambda_1, lambda_2 are consistent with y_Q.
-            // fixed_y_q * (2 * fixed_y_q - Y_{A,0}) = 0
-            let init_y_q_check = fixed_y_q.clone() * (two.clone() * fixed_y_q - Y_A_cur.clone());
```

Comment:
This change (splitting the combined selector into a selector and an advice cell) adds an extra row to every `hash_to_point` region. Instead, we should split it into a selector and a fixed column, and use the previous gate approach (but gated on the selector instead of itself). The fixed columns for the two Sinsemilla instances can then reuse the first two Lagrange coefficient columns.

Reply:
This change has been made and squashed. It has also been documented at https://github.com/zcash/orchard/pull/155/commits/78b0ec4e7bc0a712f6cf43817eb7019393ca9bc5.